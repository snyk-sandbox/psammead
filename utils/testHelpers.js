import renderer from 'react-test-renderer'; // eslint-disable-line import/no-extraneous-dependencies
import 'jest-styled-components'; // eslint-disable-line import/no-extraneous-dependencies
import ShallowRenderer from 'react-test-renderer/shallow'; // eslint-disable-line import/no-extraneous-dependencies

export const shouldMatchSnapshot = (title, component) => {
  it(title, () => {
    const tree = renderer.create(component).toJSON();
    expect(tree).toMatchSnapshot();
  });
};

export const shallowRender = component => {
  const shallowRenderer = new ShallowRenderer();
  shallowRenderer.render(component);
  const result = shallowRenderer.getRenderOutput();

  return result;
};

export const shouldShallowMatchSnapshot = (title, component) => {
  it(title, () => {
    const tree = shallowRender(component);
    expect(tree).toMatchSnapshot();
  });
};

export const isNull = (title, component) => {
  it(title, () => {
    const tree = renderer.create(component).toJSON();
    expect(tree).toBeNull();
  });
};
