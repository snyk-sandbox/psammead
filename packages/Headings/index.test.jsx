import React from "react";
import { Headline, SubHeading } from "./index";
import { shouldMatchSnapshot } from "../../utils/testHelpers";

describe("Headline component", () => {
  shouldMatchSnapshot(
    "should render correctly",
    <Headline>This is my headline.</Headline>
  );
});
describe("SubHeading component", () => {
  shouldMatchSnapshot(
    "should render correctly",
    <SubHeading text="This is a SubHeading">This is a SubHeading</SubHeading>
  );

  shouldMatchSnapshot(
    "attribute id should render without quotes",
    <SubHeading text="This 'is' a SubHeading">This is a SubHeading</SubHeading>
  );

  shouldMatchSnapshot(
    "attribute id should render without double quotes",
    <SubHeading text='This "is" a SubHeading'>This is a SubHeading</SubHeading>
  );

  shouldMatchSnapshot(
    "attribute id should render without exclamation marks",
    <SubHeading text="This is! a SubHeading!">This is a SubHeading</SubHeading>
  );
});
