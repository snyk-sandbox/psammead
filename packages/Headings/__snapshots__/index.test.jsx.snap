// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Headline component should render correctly 1`] = `
.c0 {
  color: #222222;
  font-family: ReithSerifNewsMedium,Helvetica,Arial,sans-serif;
  margin: 0;
  padding: 2rem 0 1rem 0;
  font-size: 1.75em;
  line-height: 2rem;
}

@media (min-width:20em) and (max-width:37.4375em) {
  .c0 {
    font-size: 2em;
    line-height: 2.25rem;
  }
}

@media (min-width:37.5em) {
  .c0 {
    font-size: 2.75em;
    line-height: 3rem;
  }
}

<h1
  className="c0"
>
  This is my headline.
</h1>
`;

exports[`SubHeading component attribute id should render without double quotes 1`] = `
.c0 {
  color: #404040;
  font-family: ReithSansNewsRegular,Helvetica,Arial,sans-serif;
  margin: 0;
  padding: 1rem 0;
  font-weight: 400;
  font-size: 1.25em;
  line-height: 1.5rem;
}

@media (min-width:20em) and (max-width:37.4375em) {
  .c0 {
    font-size: 1.5em;
    line-height: 1.75rem;
  }
}

@media (min-width:37.5em) {
  .c0 {
    font-size: 2em;
    line-height: 2.25rem;
  }
}

<h2
  className="c0"
  id="This-is-a-SubHeading"
  tabIndex="-1"
>
  This is a SubHeading
</h2>
`;

exports[`SubHeading component attribute id should render without exclamation marks 1`] = `
.c0 {
  color: #404040;
  font-family: ReithSansNewsRegular,Helvetica,Arial,sans-serif;
  margin: 0;
  padding: 1rem 0;
  font-weight: 400;
  font-size: 1.25em;
  line-height: 1.5rem;
}

@media (min-width:20em) and (max-width:37.4375em) {
  .c0 {
    font-size: 1.5em;
    line-height: 1.75rem;
  }
}

@media (min-width:37.5em) {
  .c0 {
    font-size: 2em;
    line-height: 2.25rem;
  }
}

<h2
  className="c0"
  id="This-is-a-SubHeading"
  tabIndex="-1"
>
  This is a SubHeading
</h2>
`;

exports[`SubHeading component attribute id should render without quotes 1`] = `
.c0 {
  color: #404040;
  font-family: ReithSansNewsRegular,Helvetica,Arial,sans-serif;
  margin: 0;
  padding: 1rem 0;
  font-weight: 400;
  font-size: 1.25em;
  line-height: 1.5rem;
}

@media (min-width:20em) and (max-width:37.4375em) {
  .c0 {
    font-size: 1.5em;
    line-height: 1.75rem;
  }
}

@media (min-width:37.5em) {
  .c0 {
    font-size: 2em;
    line-height: 2.25rem;
  }
}

<h2
  className="c0"
  id="This-is-a-SubHeading"
  tabIndex="-1"
>
  This is a SubHeading
</h2>
`;

exports[`SubHeading component should render correctly 1`] = `
.c0 {
  color: #404040;
  font-family: ReithSansNewsRegular,Helvetica,Arial,sans-serif;
  margin: 0;
  padding: 1rem 0;
  font-weight: 400;
  font-size: 1.25em;
  line-height: 1.5rem;
}

@media (min-width:20em) and (max-width:37.4375em) {
  .c0 {
    font-size: 1.5em;
    line-height: 1.75rem;
  }
}

@media (min-width:37.5em) {
  .c0 {
    font-size: 2em;
    line-height: 2.25rem;
  }
}

<h2
  className="c0"
  id="This-is-a-SubHeading"
  tabIndex="-1"
>
  This is a SubHeading
</h2>
`;
